### ISSUE TITLE
Albums: Standardize Release Date

### ISSUE THREAD

**Issue by docpop** [EmissarySocial/bandwagon#10]
When adding an album, there is a field for "release date". This field allows users to add any thing (ie "summer 08" or "July 8th, 2008". I think this should be standardized so that albums can later be organized chronologically.

**Comment by benpate** [comment]
Yes.  Thank you for this suggestion.  Fortunately, both Emissary and ActivityPub include proper dates for when something is published, so this is certainly doable.

I'll need to figure out how to use these dates, but still give bands an easy way to choose some custom formatting for them.  Maybe I'll just have a "formatted date" and a "searchable date".  I'll pay around with it a bit, and I'm open to suggestions :)

**Comment by delanthear** [comment]
Perhaps have the artist have a date format they can select (or allow them to add one using the normal format options) which all their dates will display on their page?  I.e. the artist can select if their dates appear yyyy-mm-dd, dd-mm-yyyy etc.

**Comment by benpate** [comment]
Yeah, I need to come up with a plan for this.. especially if we're going to do more with the calendar.

One thing I'd considered was to have an "actual date" that's separate from the "date label" -- possibly pre-populating the label using the date, then letting artists override the default with the formatting they want to use.

When I toured other band websites, sometimes they just wanted it written out with words ("February 24th") and sometimes with numbers ("24-2-2025")

When I get to the larger "Events" project, I'm going to draw up some ideas, then try to post them here before building.

**Comment by icaria36** [comment]
What about borrowing an existing date picker widget with an option to type the date manually but with different fields for DD / MM / YYYY? This is standard procedure like everywhere and avoid a metadata mess.

The date format rendered in album pages, events, etc, can be separate from the date input format if desired, and it could be a user preference. While standardizing data input is kind of urgent (albums being uploaded don't have standard dates), offering a setting to display dates in multiple formats is less of a priority imho (considering the backlog).  :)

**Comment by delanthear** [comment]
Yeah, it's really quite important to capture the date in a standard way.  People will definitely look at a list of releases and want to be able to look at it in release order, or reverse release order.  (I want to listen to their first album vs what was their last release?)

**Comment by benpate** [comment]
As a quick update, the build I'm working on now has similar widget for event dates, which is a good starting point for using better date formatting across the board.

Once I have this released (hopefully this week) then I can turn the same attention to album release dates.

And, since this issue is getting a lot of interest, that means it goes higher up in the release calendar. Democracy for the win :)

**Comment by benpate** [comment]
Please see #82 for a preview of what this looks like.  Including this in the next major release, expected in the next week or two.
